addons/mail/static/src/discuss/call/common/call_menu.js

```diff
@@ -9,6 +9,38 @@ export class CallMenu extends Component {
     setup() {
         super.setup();
         this.rtc = useState(useService("discuss.rtc"));
+        this.state = useState({
+            systrayIcon: undefined,
+        });
+        this.lastActions = [];
+        this.actionsIcons = [
+            "fa-desktop",
+            "fa-video-camera",
+            "fa-hand-paper-o",
+            "fa-deaf",
+            "fa-microphone-slash",
+        ];
+        useEffect(
+            (...actions) => {
+                for (let i = 0; i < actions.length; i++) {
+                    if (actions[i] !== this.lastActions[i]) {
+                        this.state.systrayIcon = actions[i]
+                            ? this.actionsIcons[i]
+                            : this.actionsIcons[actions.findIndex((v) => v)];
+                        break;
+                    }
+                }
+
+                this.lastActions = actions;
+            },
+            () => [
+                this.rtc.state.sendScreen,
+                this.rtc.state.sendCamera,
+                this.rtc.selfSession?.raisingHand,
+                this.rtc.selfSession?.isDeaf,
+                this.rtc.selfSession?.isSelfMuted,
```

Comment:
Would be nice to enforce action/state/icons better, e.g. by defining call actions with their icons too.
This would ensure same use of icon of a call action for many parts of UI, here whether call view or call (systray) menu, but also this would make this code scale better with new call features, e.g. "Start an Activity" / "Play a Sound" it'd suck to have to maintain these LOCs by hand.

Reply:
I agree it would make the code more maintainable.
We already have a list of actions inside the `CallActionList` component (partial to only the actions that appear on the dropdown) so we could introduce a `callActionsRegistry` that would simplify that component as well.